server/sonar-server/src/main/java/org/sonar/server/permission/ws/PermissionsWsParametersBuilder.java

```diff
@@ -100,33 +120,90 @@ public static void createTemplateParameters(NewAction action) {
   }
 
   private static void createTemplateIdParameter(NewAction action) {
-    action.createParam(org.sonarqube.ws.client.permission.PermissionsWsParameters.PARAM_TEMPLATE_ID)
+    action.createParam(PARAM_TEMPLATE_ID)
       .setDescription("Template id")
       .setExampleValue(Uuids.UUID_EXAMPLE_01);
   }
 
   private static void createTemplateNameParameter(NewAction action) {
-    action.createParam(org.sonarqube.ws.client.permission.PermissionsWsParameters.PARAM_TEMPLATE_NAME)
+    action.createParam(PARAM_TEMPLATE_NAME)
       .setDescription("Template name")
       .setExampleValue("Default Permission Template for Projects");
   }
 
   public static void createTemplateProjectKeyPatternParameter(NewAction action) {
-    action.createParam(org.sonarqube.ws.client.permission.PermissionsWsParameters.PARAM_PROJECT_KEY_PATTERN)
+    action.createParam(PARAM_PROJECT_KEY_PATTERN)
       .setDescription("Project key pattern. Must be a valid Java regular expression")
       .setExampleValue(".*\\.finance\\..*");
   }
 
   public static void createTemplateDescriptionParameter(NewAction action) {
-    action.createParam(org.sonarqube.ws.client.permission.PermissionsWsParameters.PARAM_DESCRIPTION)
+    action.createParam(PARAM_DESCRIPTION)
       .setDescription("Description")
       .setExampleValue("Permissions for all projects related to the financial service");
   }
 
   public static void createIdParameter(NewAction action) {
-    action.createParam(org.sonarqube.ws.client.permission.PermissionsWsParameters.PARAM_ID)
+    action.createParam(PARAM_ID)
       .setRequired(true)
       .setDescription("Id")
       .setExampleValue("af8cb8cc-1e78-4c4e-8c00-ee8e814009a5");
   }
+
+  public static NewParam createQualifierParameter(NewAction action, QualifierParameterContext context) {
+    return action.createParam(PARAM_QUALIFIER)
+      .setDescription("Project qualifier. Possible values are:" + buildRootQualifiersDescription(context))
```

Comment:
Description could be improved with something like "Filters the results with the specified qualifier" or "Indicates which type of projects to return".
Maybe a reference to Views and Cockpit plugin would be useful too.